### PR TITLE
fix: iOS not opening

### PR DESCRIPTION
**Pull request by JDLegaspi**
iOS does not open because the ThanksWidget does not unmount, and doesn't update. 

Opening the iOS `<Widget>`, closing it, then trying to open it again results in `<Widget>` not opening. For now, we add an `isOpen` state so that the component is not rendered when this is `false`.

The component update will be made within the `rocket` repo.

### CODE REVIEW

**Comment by theKashey** [comment]
🤔 What you are solving here?

**Comment by JDLegaspi** [comment]
@theKashey Added changed the description to be a bit more descriptive. When opening Widget, then closing it, it will not open a second time. 

This is only the case for the DirectWidget.

File: react-native/app.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    version: "1.0.0",
+    orientation: "portrait",
+    userInterfaceStyle: "light",
+    splash: {
+      resizeMode: "contain",
+      backgroundColor: "#ffffff",
+    },
+    assetBundlePatterns: ["**/*"],
+    plugins: [
+      path.resolve(__dirname, "./src/plugins/withReactNativeWebview.js"),
```
Comment (JDLegaspi) [inline react-native/app.config.js:16]: Added this in order to support Native Modules. However, after exploring a different Web View version in `@thanksjs/react-native-webview`, we may be able to get rid of this.

As you said in the other PR @theKashey, a 'real' RN developer should already know this.

File: react-native/package.json [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
+  "scripts": {
+    "ios": "expo run:ios",
+    "start": "expo start -c",
+    "android": "expo run:android"
+  },
+  "dependencies": {
+    "@babel/preset-typescript": "^7.23.2",
+    "@thanksjs/react-native-webview": "^1.1.0",
+    "@types/react": "~18.2.45",
+    "expo": "^51.0.31",
```
Comment (JDLegaspi) [inline react-native/package.json:15]: Updated Expo to support the current Expo Go version

File: react-native/src/App.tsx [hunk 3]
```diff
@@ -1,54 +1,80 @@
...
+          }}
+        />
+        <Button title="display indirect example" onPress={() => setMode(2)} />
+        <Text>mode is: {["none", "simple", "indirect"][mode]}</Text>
+        {new Array(20).fill(0).map((_, i) => (
+          <Text key={i} style={{ color: "#FFF" }}>
+            {new Array(4).fill(" Thanks ").join("")}
+          </Text>
+        ))}
+        {mode === 1 && isOpen && <DirectWidget onClose={handleClose} />}
```
Comment (JDLegaspi) [inline react-native/src/App.tsx:75]: Conditional render will allow the `DirectWidget` to open again once it's closed

**Review by theKashey — APPROVED** [review]

**Pull request merged**